app/controllers/instructeurs/dossiers_controller.rb

```diff
@@ -273,16 +273,16 @@ def create_avis
     end
 
     def update_annotations
-      dossier_with_champs.assign_attributes(champs_private_params)
-      if dossier.champs_private_all.any?(&:changed?)
+      dossier_with_champs.update_champs_private(champs_private_params[:champs_private_all_attributes])
+      if dossier.champs.any?(&:changed_for_autosave?) || @dossier.champs_private_all.any?(&:changed_for_autosave?) # TODO remove second condition after one deploy
```

Comment:
je retire le `@` , et reviens a `changed` pourquoi le changer ? pourquoi checker aussi les champs publiques ?

```suggestion
      if dossier.champs.any?(&:changed?) || dossier.champs_private_all.any?(&:changed?) # TODO remove second condition after one deploy
```